### PR TITLE
add a GMA nature, and operate GMA transformations only on GMA projects

### PR DESCRIPTION
**Pull request by PhilippeNoel**
Bonjour Olivier,
confronté à de multiple projets je me trouve utiliser certains avec GenModelAddon et d'autres où ce n'est pas nécessaire.
J'en suis venu à ajouter une nature, et à n'appliquer les transformations ou la génération de la structure seulement si le projet contenant le modèle a la bonne nature.
Ajouté aussi la commande permettant d'ajouter ou supprimer la nature en question sur un projet.

Par contre, je dois encore faire en sorte que les tests en tiennent compte, ce qui n'est pas le cas encore aujourd'hui

Philippe Noel

### CODE REVIEW

**Comment by opcoach** [comment]
Hey ! Salut Philippe ! Et bonne année ! ;) 

I will switch the comment to English ;) 

This is a good idea to have this nature on the project. It is important to distinguish the code generation depending on the project (need GMA generation or not). 

Have you seen that the GenModelFactory is overridden by the genModelAddon.core plugin ? It means that all objects created when reading the genmodel file are created by this factory. The reason is to simplify the way to compute the class names by the default EMF generator and to avoid to change any code in the default EMF generator ! 

Inside GMAGenXXXX instances, some methods are overridden to call the GMATransform.replaceDevName static method. (see the end of file : https://github.com/opcoach/genModelAddon/blob/master/com.opcoach.genmodeladdon.core/src/com/opcoach/genmodeladdon/core/genmodel/GMATransform.xtend)

May be this method should transform the names not only if this is a GMAGenModel instance but also if this model is inside a GMA nature project ?   

What will happen if we select/unselect the GMA nature if the genModel objects instances are already created, and the we regenerate the EMF model code layer  ?  Have you tested it ? 

Did you also consider that a '.gma' file is created in the root directory of the project ? It contains the properties setup from the GMA dialog so as to restore it. Wouldn't be enough to know that it is a GMA project ? 

And to finish, depending on the GMA project nature, the 'Generate Dev Structure' in the menu should be displayed or not. So should we let the menu always displayed and switch the nature automatically, or should we first switch the nature, and then access to the menu ?  

I opened a new issue to track your work : https://github.com/opcoach/genModelAddon/issues/89

I also accepted the Actions on the GitHub project for your contribution. It seems that a push in your fork should be automatically compiled on your side (so as to check if all the tests are passed). 

Discussion is opened :)

File: com.opcoach.genmodeladdon.core/src/com/opcoach/genmodeladdon/core/GMAConstants.xtend [hunk 1]
```diff
@@ -34,6 +34,6 @@ interface GMAConstants {
 	public static val String DEFAULT_SRC_CLASS_IMPL_PATTERN = "{0}DevImpl"
 	
 	// GMA nature
-  	public static val String NATURE_ID = "com.opcoach.genmodeladdon.core.GMANature";
+  	public static val String NATURE_ID = "com.opcoach.genmodeladdon.ui.GMANature";
```
Comment (opcoach) [inline com.opcoach.genmodeladdon.core/src/com/opcoach/genmodeladdon/core/GMAConstants.xtend:37]: Should be core not ui (we are in a core plugin)

File: com.opcoach.genmodeladdon.core/xtend-gen/com/opcoach/genmodeladdon/core/GMAConstants.java [hunk 2]
```diff
@@ -36,5 +36,5 @@
   
   static final String DEFAULT_SRC_CLASS_IMPL_PATTERN = "{0}DevImpl";
   
-  static final String NATURE_ID = "com.opcoach.genmodeladdon.core.GMANature";
+  static final String NATURE_ID = "com.opcoach.genmodeladdon.ui.GMANature";
```
Comment (opcoach) [inline com.opcoach.genmodeladdon.core/xtend-gen/com/opcoach/genmodeladdon/core/GMAConstants.java:39]: core, not ui

Comment (PhilippeNoel) [reply com.opcoach.genmodeladdon.core/xtend-gen/com/opcoach/genmodeladdon/core/GMAConstants.java:39]: I Agree, then we must change nature Id inside ui plugin.xml for contribution menus (add/remove nature)

**Comment by opcoach** [comment]
Fix #89 : see comment inside.

**Comment by opcoach** [comment]
Close #89

**Pull request merged**